docs/developer/payments/transactions.mdx

```diff
@@ -450,6 +450,77 @@ Once `Checkout` is converted to into `Order`, the `authorizeStatus` might change
 - **For orders**: pending Transactions **are not** considered "paid" and will not be included in the [`Order.authorizeStatus`](/api-reference/orders/enums/order-authorize-status-enum.mdx#values) calculation.
 :::
 
+### TransactionItem and  TransactionEvents pspReference Matching
+
+Both [TransactionItem](/api-reference/payments/objects/transaction-item)
+and [TransactionEvent](/api-reference/payments/objects/transaction-event)
+contain a `pspReference` field, which refers to the payment service provider reference.
+	-	The `pspReference` on `TransactionItem` identifies the overall payment.
+	-	The `pspReference` on `TransactionEvent` identifies specific actions that occurred within the `Transaction`.
+
+**Key Assumptions:**
+- Only one `TransactionItem` instance can exist with a given `pspReference`.
+- For a specific `TransactionItem`, only one `TransactionEvent` of a given `type` and `pspReference` may exist, except for the following types:
+  -	`AUTHORIZATION_ACTION_REQUIRED`
+	-	`CHARGE_ACTION_REQUIRED`
+	-	`INFO`
+- Only one `TransactionEvent` of type `AUTHORIZATION_SUCCESS` is allowed to exist.
+
+#### Setting the `pspReference` on the `TransactionItem`
+
+The pspReference value on a `TransactionItem` can initially be `None`.
+When the first `TransactionEvent` containing a `pspReference` is reported,
+this value is automatically set on the associated `TransactionItem`.
+
+The `pspReference` can be updated through the `transactionUpdate` mutation if necessary.
+
+#### Events with optional `pspReference` 
+
+The following events do not require a `pspReference`:
+- `CHARGE_ACTION_REQUIRED`
+- `AUTHORIZATION_ACTION_REQUIRED`
+- `CHARGE_FAILURE`
+- `AUTHORIZATION_FAILURE`
+- `REFUND_FAILURE`
+- `CANCEL_FAILURE`
+
+If `pspReference` is missing, the event is not considered in transaction amount recalculations.
```

Comment:
I can extend it a little, but I do not want to describe here how the recalculations are woking because it should be separate section just linked here.